### PR TITLE
Studio

### PR DESCRIPTION
**Pull request by Leland-Takamine**
Adding example that includes an `index` when there are multiple elements matching.

image

### CODE REVIEW

File: maestro-studio/server/src/main/java/maestro/studio/MaestroStudio.kt [hunk 1]
```diff
@@ -103,16 +107,39 @@ object MaestroStudio {
...
+                gatherElements(child, list)
+            }
+            list.add(tree)
+            return list
+        }
+        fun TreeNode.attribute(key: String): String? {
+            val value = attributes[key]
+            if (value.isNullOrEmpty()) return null
+            return value
+        }
```
Comment (axelniklasson) [inline maestro-studio/server/src/main/java/maestro/studio/MaestroStudio.kt:122]: **Big** nit: Should this extension function be defined outside of `treeToElements` next to `TreeNode.bounds()`? Do all the defined functions in `treeToElements` get recreated on every invocation?

File: maestro-studio/web/src/Examples.tsx [hunk 2]
```diff
@@ -42,13 +42,24 @@ const Section = ({ deviceScreen, element, title, documentationUrl, codeSnippets
...
     if (codeSnippet.includes('tapOn') && isBoundsEmpty(element)) return null
+
+    if (elementHasResourceIdIndex && codeSnippet.includes('[id]') && !codeSnippet.includes('[resource-id-index]')) return null
+    if (elementHasTextIndex && codeSnippet.includes('[text]') && !codeSnippet.includes('[text-index]')) return null
+
     const id = element.resourceId || ''
     const text = element.text || ''
+    const resourceIdIndex = `${element.resourceIdIndex}`
+    const textIndex = `${element.textIndex}`
+    console.log(element)
```
Comment (axelniklasson) [inline maestro-studio/web/src/Examples.tsx:62]: Nit: remove `console.log`

File: maestro-studio/web/src/Examples.tsx [hunk 3]
```diff
@@ -42,13 +42,24 @@ const Section = ({ deviceScreen, element, title, documentationUrl, codeSnippets
...
   const codeSnippetComponents = codeSnippets.map(codeSnippet => {
     if (codeSnippet.includes('[id]') && !element.resourceId) return null
+    if (codeSnippet.includes('[resource-id-index]') && !elementHasResourceIdIndex) return null
+    if (codeSnippet.includes('[text-index]') && !elementHasTextIndex) return null
     if (codeSnippet.includes('[text]') && !element.text) return null
     if (codeSnippet.includes('[bounds]') && !element.bounds) return null
     if (codeSnippet.includes('tapOn') && isBoundsEmpty(element)) return null
+
+    if (elementHasResourceIdIndex && codeSnippet.includes('[id]') && !codeSnippet.includes('[resource-id-index]')) return null
+    if (elementHasTextIndex && codeSnippet.includes('[text]') && !codeSnippet.includes('[text-index]')) return null
```
Comment (axelniklasson) [inline maestro-studio/web/src/Examples.tsx:56]: Observation: this is getting a little hard to follow along -- is there a way we can make this easier to read? Maybe splitting it out into separate functions with descriptive names, adding comments, or something else?

**Review by axelniklasson — APPROVED** [review]

File: maestro-studio/web/src/Examples.tsx (hunk 2 above)
Comment (Leland-Takamine) [reply maestro-studio/web/src/Examples.tsx:62]: Good catch!

File: maestro-studio/web/src/Examples.tsx (hunk 3 above)
Comment (Leland-Takamine) [reply maestro-studio/web/src/Examples.tsx:56]: Let me add some comments in here

**Review by dmitry-zaitsev — APPROVED** [review]
